### DISCUSSION TITLE
how to override the createWebSocketUrl method in extension

### DISCUSSION THREAD

**Discussion by LionLion001** [eclipse-theia/theia#11959] [Q&A] [answered]
i create an extension to overrides the createWebSocketUrl method ,but it does not work .Did I make a mistake？
https://github.com/XiaoXuan001/ws

**Comment by msujew** [comment, marked as answer]
Hey @XiaoXuan001,

thanks for the repro code. The override for the `WebSocketConnectionProvider` shouldn't be in the `node` part of your application but rather in your `browser` part. See also the [path of the original class](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/messaging/ws-connection-provider.ts).

**Reply by LionLion001** [reply]
thanks,when i change it to browser, it gives such a mistake
![image](https://user-images.githubusercontent.com/31761281/205891608-d9fd115a-f982-45ab-b687-7b3c9acd2495.png)

**Reply by msujew** [reply]
Ah, yes. Instead of just binding the identifier using `bind` you need to call `rebind` to actually change the target for the injection.
